src/views/viewer/EditMembershipView/EditMembershipView.tsx

```diff
@@ -0,0 +1,192 @@
+import { useApolloClient } from '@apollo/client'
+import { zodResolver } from '@hookform/resolvers/zod'
+import debouncePromise from 'awesome-debounce-promise'
+import React, { useCallback, useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import useMeasure from 'react-use-measure'
+import * as z from 'zod'
+
+import { GetMembershipDocument, GetMembershipQuery, GetMembershipQueryVariables } from '@/api/queries'
+import { LimitedWidthContainer } from '@/components/LimitedWidthContainer'
+import { MembershipInfo } from '@/components/MembershipInfo'
+import { TextArea } from '@/components/_inputs/TextArea'
+import { MEMBERSHIP_NAME_PATTERN, URL_PATTERN } from '@/config/regex'
+import { useUser } from '@/providers/user'
+
+import { StyledActionBar, StyledTextField, TextFieldsWrapper, Wrapper } from './EditMembershipView.styles'
+
+type Inputs = {
+  handle: string
+  avatar: string | null
+  about: string | null
+}
+
+export const EditMembershipView: React.FC = () => {
+  const { activeAccountId, activeMembership, activeMembershipLoading } = useUser()
+  const [actionBarRef, actionBarBounds] = useMeasure()
+
+  const client = useApolloClient()
+
+  const avatarValidation = async (value: string): Promise<string | boolean> =>
+    new Promise((resolve) => {
+      const image = new Image()
+      image.onload = () => {
+        resolve(true)
+      }
+      image.onerror = () => resolve(false)
+      image.src = value
+      if (!value) {
+        resolve(true)
+      }
+    })
+
```

Comment:
This can live outside the component and in fact, utils would probably be a good place for it

Reply:
This may be like a 3rd place in the app where we use something similar, definitely agree